sds_data_manager/lambda_code/SDSCode/database/models.py

```diff
@@ -98,32 +109,79 @@ class FileCatalog(Base):
     """File catalog table"""
 
     __tablename__ = "file_catalog"
+    __table_args__ = UniqueConstraint(
+        "id",
+        "file_path",
+        "instrument",
+        "data_level",
+        "start_date",
+        "end_date",
+        name="file_catalog_uc",
+    )
 
     # TODO: determine cap for strings
     id = Column(Integer, Identity(start=1, increment=1), primary_key=True)
     file_path = Column(String, nullable=False)
-    instrument = Column(instruments, nullable=False)
-    data_level = Column(data_levels, nullable=False)
-    descriptor = Column(String, nullable=False)
+    instrument = Column(INSTRUMENTS, nullable=False)
+    data_level = Column(DATA_LEVELS, nullable=False)
+    descriptor = Column(String(20), nullable=False)
     start_date = Column(DateTime, nullable=False)
     end_date = Column(DateTime, nullable=False)
-    version = Column(String, nullable=False)
-    extension = Column(String, nullable=False)
+    version = Column(String(6), nullable=False)
```

Comment:
Chose `6` here for the 6 characters in `vXX-YY`

Reply:
Maybe make it 7 or 8 just in case? I could see hitting 3 digit versions and it would be an annoying thing to fix